### PR TITLE
Add Coingecko extension

### PR DESCRIPTION
**Pull request by Destiner**
## Description

Adds a new "Coingecko" extension. The extension allows to search crypto coins on [Coingecko](https://www.coingecko.com), get current and historical price, as well as get coin metadata (name, symbol, marketcap rank, and project's homepage). User can also add a coin to favourites.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/4247901/146766954-e94e642c-54c7-4d38-9ac7-a4e0e3339f2c.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/coingecko/package.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "currency",
+      "type": "dropdown",
+      "required": false,
+      "title": "Currency",
+      "description": "Currency that is used to diplay coin prices",
```
Comment (fe9lix) [inline extensions/coingecko/package.json:24]:
```suggestion
      "description": "Currency that is used to display coin prices",
```

**Review by fe9lix — CHANGES_REQUESTED** [review]
Hey @Destiner – thanks for your contribution!
Question: Any reason you're not taking advantage of built-in list filtering here? It doesn't look like the query is performed against a search/typeahead endpoint that would process the input query but rather the coins list is first loaded and then manually filtered. With built-in filtering you would get fuzzy matching on the title (coin name) and you could set the coin symbol as searchable keyword, for example.

**Comment by Destiner** [comment]
Hey @fe9lix, thanks for feedback!

I'd be happy to use built-in filtering provided by Raycast. The problem is, the number of entries returned by the Coingecko API is >10k. When I try to pass the array of those entries into `<List>`, the extension often throws due to `Out Of Memory` error. I assumed this is due to huge number of items. I had to limit the initial amount of items rendered to 1000, but still allow to search all 10'000 items.

Do you have an idea on how to get rid of that custom filtering logic while avoiding rendering too much items?

**Comment by fe9lix** [comment]
Okay, I see. Yeah, the list shouldn't really contain more then 1000 items (although we do have extensions in the store that almost show ~3k items and work okayish performance-wise). Maybe one thing worth improving in that case is to allow contains-like search rather than using startsWith() – or would that cause performance issues?

**Comment by Destiner** [comment]
Cool, I replaced `startsWith` with `includes`. Performance doesn't seem to be affected much, filtering is still pretty fast.

**Review by fe9lix — APPROVED** [review]

**Pull request merged**